### PR TITLE
fix the build if MPI is not available

### PR DESCRIPTION
**Pull request by andlaus**
I'm not really sure if the fix is semantically correct (Dune
communicators are strange beasts if you're new to them), but at least
this patch fixes the build without MPI and the resulting flow_ebos
binary seems to work fine.

I'll self-merge this immediately as it is rather simple and fixes an issue encountered in the real world by @akva2. Patches for fixing this  in a better way are highly welcome!

### CODE REVIEW

File: opm/autodiff/BlackoilModelEbos.hpp [hunk 1]
```diff
@@ -439,7 +442,11 @@ namespace Opm {
           typedef Y range_type;
           typedef typename X::field_type field_type;
 
+#if HAVE_MPI
           typedef Dune::OwnerOverlapCopyCommunication<int,int> communication_type;
+#else
+          typedef Dune::CollectiveCommunication<int> communication_type;
```
Comment (blattms) [inline opm/autodiff/BlackoilModelEbos.hpp]: Theses two types are not semantically related as they have totally different interfaces.

**Comment by andlaus** [comment]
> In a parallel run you would wrap this class in an OverlappingSchwarzOperator and get the project stuff for free. No need to fiddle with #if MPI in this class.

gimme a patch ;)

File: opm/autodiff/BlackoilModelEbos.hpp (hunk 1 above)
Comment (andlaus) [inline opm/autodiff/BlackoilModelEbos.hpp]: I though so. their naming is quite bad for this reason, IMO.

**Comment by blattms** [comment]
Did not find one. My desk is a mess...

**Comment by andlaus** [comment]
**g** let's stay with the current kludge then...

**Pull request merged**